lib/cmds/space_cmds/generate_cmds/migration.js

```diff
@@ -0,0 +1,237 @@
+
+  const chain = Object.keys(_.omit(field, 'id')).map((key) => {
+    return [key, field[key]]
+  })
+
+  return createCallChain(
+    b.callExpression(
+      b.memberExpression(
+        b.identifier(ctVariable),
+        b.identifier('createField')
+      ),
+      [b.literal(fieldId)]
+    ),
+    chain
+  )
+}
+
+const changeEditorInterface = function (ctId, fieldId, widgetId, settings) {
+  const ctVariable = ctVariableEscape(ctId)
+  settings = settings || {}
+
+  const settingsExpression = b.objectExpression(
+    _.map(settings, (v, k) => {
+      return b.property('init', b.identifier(k), b.literal(v))
+    })
+  )
+
+  return b.callExpression(
+    b.memberExpression(
+      b.identifier(ctVariable),
+      b.identifier('changeEditorInterface')
+    ),
+    [b.literal(fieldId), b.literal(widgetId), settingsExpression]
+  )
+}
+
+const getContentTypes = async function (environment, contentTypeId) {
+  return contentTypeId ? [await environment.getContentType(contentTypeId)] : (await environment.getContentTypes()).items
+}
+
+const generateContentTypeMigration = async function (environment, contentType) {
+  log(`Creating migration for content type: '${contentType.sys.id}'`)
+
+  const fieldCreators = contentType.fields.map((field) => {
+    return b.expressionStatement(createField(contentType.sys.id, field))
+  })
+
+  let editorInterfacesCreators = []
+  try {
+    log('Fetching editor interface')
+
+    const editorInterface = await environment.getEditorInterfaceForContentType(contentType.sys.id)
+    editorInterfacesCreators = contentType.fields.map((field) => {
+      const control = _.find(editorInterface.controls, (control) => {
+        return control.fieldId === field.id
+      })
+      const widgetId = control.widgetId
+      const settings = control.settings
+
+      return b.expressionStatement(changeEditorInterface(contentType.sys.id, field.id, widgetId, settings))
+    })
+  } catch (err) {
+    // We do nothing - This means that the Content Type has no fields
+  }
+
+  return [createContentType(contentType)].concat(fieldCreators).concat(editorInterfacesCreators)
+}
+
+const generateMigrationScript = async function (environment, contentTypes) {
+  const migrationContents = await Promise.all(_.map(contentTypes, async (contentType) => {
+    return generateContentTypeMigration(environment, contentType)
+  }))
+  const migration = wrapMigrationWithBase(b.blockStatement(_.flatten(migrationContents)))
+
+  const output = recast.prettyPrint(migration, { tabWidth: 2 }).code
+  return prettier.format(output)
+}
+
+const generateFileName = function (spaceId, environmentId, contentTypeId) {
+  const contentTypePart = contentTypeId ? `-${contentTypeId}` : ''
+
+  return `${spaceId}-${environmentId}${contentTypePart}-${Date.now()}.js`
+}
+
+export async function generateMigration (argv) {
+  await assertLoggedIn()
+  await assertSpaceIdProvided(argv)
+  await assertEnvironmentIdProvided(argv)
```

Comment:
IMO we can remove the `assertEnvironmentIdProvided`.
for the environment case, we can default it to `master` if it's not provided like the import and export is doing. This will provide a better experience.